File: integration-tests-java/src/main/java/io/micronaut/jimmer/java/it/config/ConsumerZone.java
package io.micronaut.jimmer.java.it.config;

import io.micronaut.context.annotation.Factory;
import jakarta.inject.Named;
import jakarta.inject.Singleton;
import java.time.ZoneId;
import java.util.function.Consumer;
import org.babyfish.jimmer.sql.JSqlClient;

@Factory
public class ConsumerZone {

    @Singleton
    @Named(Constant.DEFAULT)
    public Consumer<JSqlClient.Builder> jSqlClientZoneIdConfigurerForDefaultDb() {
        return builder -> builder.setZoneId(ZoneId.of("Asia/Shanghai"));
    }

    @Singleton
    @Named(Constant.DATASOURCE2)
    public Consumer<JSqlClient.Builder> jSqlClientZoneIdConfigurerForDb2() {
        return builder -> builder.setZoneId(ZoneId.of("Asia/Chongqing"));
    }
}

File: integration-tests-kotlin/src/main/kotlin/io/micronaut/jimmer/kotlin/it/config/ConsumerZone.kt
package io.micronaut.jimmer.kotlin.it.config

import io.micronaut.context.annotation.Factory
import jakarta.inject.Named
import jakarta.inject.Singleton
import org.babyfish.jimmer.sql.JSqlClient
import java.time.ZoneId
import java.util.function.Consumer

@Factory
class ConsumerZone {
    @Singleton
    @Named(Constant.DEFAULT)
    fun jSqlClientZoneIdConfigurerForDefaultDb(): Consumer<JSqlClient.Builder> =
        Consumer { builder: JSqlClient.Builder ->
            builder.setZoneId(ZoneId.of("Asia/Shanghai"))
        }

    @Singleton
    @Named(Constant.DATASOURCE2)
    fun jSqlClientZoneIdConfigurerForDb2(): Consumer<JSqlClient.Builder> =
        Consumer { builder: JSqlClient.Builder ->
            builder.setZoneId(ZoneId.of("Asia/Chongqing"))
        }
}
